types/wlr_surface.c

```diff
@@ -805,7 +805,7 @@ static void subsurface_handle_place_above(struct wl_client *client,
 	}
 
 	wl_list_remove(&subsurface->parent_pending_link);
-	wl_list_insert(sibling->parent_pending_link.prev,
+	wl_list_insert(&sibling->parent_pending_link,
```

Comment:
Wait, I think the `subsurface_handle_place_above` code was correct before. `place_above` means "place it above in rendering order" right?

Reply:
I naively thought above = will be drawn after in rendering order so it shows "on top" = inserted after in the list since we're walking it up the normal order.

Let's see...
```
    <request name="place_above">
      <description summary="restack the sub-surface">
        This sub-surface is taken from the stack, and put back just
        above the reference surface, changing the z-order of the sub-surfaces.
        The reference surface must be one of the sibling surfaces, or the
        parent surface. Using any other surface, including this sub-surface,
        will cause a protocol error.

        The z-order is double-buffered state, and will be applied on the
        next commit of the parent surface.
        See wl_surface.commit and wl_subcompositor.get_subsurface.

        A new sub-surface is initially added as the top-most in the stack
        of its siblings and parent.
      </description>

      <arg name="sibling" type="object" interface="wl_surface"
           summary="the reference surface"/>
    </request>
```
Not helpful for me, neither is the rest of the xml file after a very quick glance through, how do you understand that?

Reply:
Our list is in the rendering order which is "natural", ie. the first item is at the bottom and the last item is at the top. So this request should place the subsurface right after the reference. `wl_list_insert(a, b)` inserts `b` right after `a`. So we should be fine if we `wl_list_insert(&sibling->parent_pending_link, &subsurface->parent_pending_link)`, ie. your code is correct?